src/renderers/shaders/ShaderChunk/common.glsl

```diff
@@ -4,57 +4,136 @@
 #define LOG2 1.442695
 #define EPSILON 1e-6
 
-float square( in float a ) { return a*a; }
-vec2  square( in vec2 a )  { return vec2( a.x*a.x, a.y*a.y ); }
-vec3  square( in vec3 a )  { return vec3( a.x*a.x, a.y*a.y, a.z*a.z ); }
-vec4  square( in vec4 a )  { return vec4( a.x*a.x, a.y*a.y, a.z*a.z, a.w*a.w ); }
+float square( in float a ) { return a * a; }
+vec2  square( in vec2 a )  { return vec2( a.x * a.x, a.y * a.y ); }
+vec3  square( in vec3 a )  { return vec3( a.x * a.x, a.y * a.y, a.z * a.z ); }
+vec4  square( in vec4 a )  { return vec4( a.x * a.x, a.y * a.y, a.z * a.z, a.w * a.w ); }
+
```

Comment:
Unrelated to the phong shader changes, but these functions are a bit weird as its just ````return a * a```` in all cases? Also using them may cause a perf loss if you are also doing an add instruction as you could loos out on a fused MAD (multiply+add) instruction.

e.g. ````a * a + b```` is one instruction but ````square(a) + b```` is many especially if its doing per component multiplys in the function.